src/utilities/theme/utils.ts

```diff
@@ -1,115 +1,779 @@
-import tokens from '@shopify/polaris-tokens';
-import {HSLColor} from '../color-types';
+import {HSLColor, HSLAColor} from '../color-types';
 import {colorToHsla, hslToString, hslToRgb} from '../color-transformers';
 import {isLight} from '../color-validation';
-import {constructColorName} from '../color-names';
-import {createLightColor} from '../color-manipulation';
-import {compose} from '../compose';
-
-import {Theme, ColorsToParse, ThemeVariant, ThemeColors} from './types';
-import {needsVariantList} from './config';
-
-export function setColors(theme: Theme | undefined): string[][] | undefined {
-  let colorPairs;
-  if (theme && theme.colors) {
-    Object.entries(theme.colors).forEach(([colorKey, pairs]) => {
-      const colorKeys = Object.keys(pairs);
-      if (colorKey === 'topBar' && colorKeys.length > 1) {
-        colorPairs = colorKeys.map((key: string) => {
-          const colors = (theme.colors as ThemeColors).topBar;
-          return [constructColorName(colorKey, key), colors[key]];
-        });
-      } else {
-        colorPairs = parseColors([colorKey, pairs]);
-      }
-    });
-  }
+import {CSSProperties, Theme} from './types';
 
-  return colorPairs;
+// TODO: Pull from polaris-tokens
+enum DefaultColor {
+  Surface = '#FAFAFA',
+  OnSurface = '#1F2225',
+  InteractiveNeutral = '#EAEAEB',
+  Interactive = '#0870D9',
+  Branded = '#008060',
+  Critical = '#E32727',
+  Warning = '#FFC453',
+  Highlight = '#59D0C2',
+  Success = '#008060',
 }
 
-export function needsVariant(name: string) {
-  return needsVariantList.indexOf(name) !== -1;
-}
+export function Colors(theme: Theme) {
+  const {colors = {}} = theme;
+
+  const isLightTheme = isLight(
+    hslToRgb(colorToHsla(
+      colors == null || colors.surface == null
+        ? DefaultColor.Surface
+        : colors.surface,
+    ) as HSLColor),
+  );
+
+  const {
+    surface = DefaultColor.Surface,
+    onSurface = DefaultColor.OnSurface,
+    interactive = DefaultColor.Interactive,
+    branded = DefaultColor.Branded,
+    critical = DefaultColor.Critical,
+    warning = DefaultColor.Warning,
+    highlight = DefaultColor.Highlight,
+    success = DefaultColor.Success,
+  } = colors;
 
-const lightenToString: (
-  color: HSLColor | string,
-  lightness: number,
-  saturation: number,
-) => string = compose(
-  hslToString,
-  createLightColor,
-);
-
-export function setTextColor(
-  name: string,
-  variant: ThemeVariant = 'dark',
-): string[] {
-  if (variant === 'light') {
-    return [name, tokens.colorInk];
+  return customPropertiesTransformer({
+    ...surfaceColors(surface),
+    ...onSurfaceColors(onSurface),
+    ...interactiveColors(interactive),
+    ...interactiveNeutralColors(onSurface),
+    ...brandedColors(branded),
+    ...criticalColors(critical),
+    ...warningColors(warning),
+    ...highlightColors(highlight),
+    ...successColors(success),
+  });
```

Comment:
This builds up a flat object of CSS Custom Properties containing the entire color system generated from the base values.